.github/workflows/test.yml

```diff
@@ -1,10 +1,15 @@
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
 name: Test
 jobs:
   test:
     strategy:
       matrix:
-        go-version: [1.19.x]
+        go-version: [ '1.19.x', '1.20.x' ]
```

Comment:
Our code is just POGO (like we say POJO in Java world). We are doing nothing fancy like FFI into C or some OS dependent work etc. And given Go's strong backwards compatibility promise it is very likely that our code would run on future versions so why waste precious resources on extra CI jobs :) So just testing against 1.20 is fine IMO

Reply:
sorry I meant 1.19

Reply:
Why 1.19 and not the latest 1.20?

Reply:
Okay first of all it seems like I forgot to downgrade module version from 1.20 to 1.19 while making final PR at the time. By default when I initialized go module it used whatever go version was on the dev machine i.e. 1.20.

But yes you can also fix that in this PR: change go version to 1.19 in go.mod.

Q: Why a library package like this should not be on the latest versions?
A: So that consumers on older versions can also use our package. Basically the lower version you go the larger audience you can offer your package to. See these well known repositories as examples. These are actively maintained yet they are not on the latest Go versions:
- https://github.com/go-chi/chi/blob/master/go.mod
- https://github.com/google/go-github/blob/master/go.mod

But going too low you miss out on new features, bug fixes etc. so generally such packages stay 2-5 versions back if they know what they are doing. For our code we can even if we go as low as 1.16 and it would be all good except some features would need to go away because for them minimum go version is 1.19.

Now of course we can test our code against latest versions in CI to get the confidence that we can run on newer versions of Go as well. This is okay and we can add as many Go versions in test matrix as we like. But given Go's backward compatibility record and we not doing anything fancy/complicated in our code base,  I do not see any value in adding too many CI jobs to waste our mother Earth's resources just so that our dashboards look pretty. However once 1.21 is released we will add it to the matrix so that we can say that we run correctly on our base version and on latest version and that should be enough.